server/pbench/bin/pbench-server-prep-shim-001

```diff
@@ -0,0 +1,220 @@
+    echo "Failed: \"getconf.py archive-dir pbench-server" >> $errlog
+    exit 1
+fi
+
+if [[ "$quarantine" == "" ]] ;then
+    echo "Failed: \"getconf.py quarantine-dir pbench-server" $errlog
+    exit 1
+fi
+
+if [[ "$duplicates" == "" ]] ;then
+    echo "Failed: \"getconf.py duplicates-dir pbench-server" $errlog
+    exit 1
+fi
+
+log_init $(basename $0) $LOGSDIR/$(basename $0)
+
+tmp=$TMP/$PROG.$$
+
+function quarantine () {
+    dest=$1
+    shift
+    files="$@"
+
+    mkdir -p $dest || return $?
+    mv $files $dest || return $?
+}
+
+trap 'rm -rf $tmp' EXIT INT QUIT
+
+mkdir -p $tmp
+list=$tmp/list.tb
+status=$tmp/status
+> $status
+# we are explicitly handling version-001 data
+results_dir=$(getconf.py receive-dir-${version} pbench-server)
+
+# check for results that are ready for processing: version 001
+# agents create a link in the TODO subdir (which is created if
+# necessary) after they finish with MD5 checking, so that's what
+# we look for.
+find ${results_dir} -maxdepth 3 -type l -name '*.tar.xz' > $list
+
+typeset -i ntbs=0
+typeset -i nerrs=0
+typeset -i nquarantined=0
+typeset -i ndups=0
+
+echo $TS
+while read tblink ;do
+    # resolve the link
+    tb=$(readlink -f ${tblink})
+    sts=$?
+    if [ $sts != 0 ] ;then
+        echo "$TS: readlink failed: ${tblink}" | tee -a $status >&4
+        nerrs=$nerrs+1
+        continue
+    fi
+
+    # directory
+    tbdir=$(dirname ${tb})
+
+    # resultname: get the basename foo.tar.xz and then strip the .tar.xz
+    resultname=$(basename ${tb})
+    resultname=${resultname%.tar.xz}
+
+    # hostname is the last component of the directory part of the full path
+    controller=$(basename ${tbdir})
+
+    # TBD
+    dest=$archive/$controller
+
+    # sanity checks
+    # Impossible: the tarball disappeared between the find and this check!
+    if [ ! -f ${tb} ] ;then
+        echo "$TS: ${tb} file does not exist" | tee -a $status >&4
+        nerrs=$nerrs+1
+        continue
+    fi
+
+    if [ ! -f ${tb}.md5 ] ;then
+        # this may be the result of pbench-move-results racing with this script:
+        # we'll just skip the tarball this time and try again next time.
+        echo "$TS: ${tb}.md5 file does not exist" >&4
+        continue
+    fi
+
+    # This should not happen: version 001 agents check for duplicates in pbench-move-results.
+    if [ -f ${dest}/${resultname}.tar.xz -o -f ${dest}/${resultname}.tar.xz.md5 ] ;then
+        quarantine $duplicates/$controller  ${tb} ${tb}.md5
+        sts=$?
+        if [ $sts -ne 0 ] ;then
+            echo "Failed: quarantine $quarantine/$controller ${tb}, status $sts" |
+                tee -a $status >&4
+        fi
+        ndups=$ndups+1
+    fi
+
+    pushd ${tbdir} >/dev/null 2>&1
+    # MD5 check
+    md5sum --check $(basename $tb).md5 
+    sts=$?
+    if [ $sts -ne 0 ] ;then
+        echo "$TS: ${tb} failed MD5 check" >&4
+        quarantine $quarantine/$controller ${tb} ${tb}.md5
+        sts=$?
```

Comment:
Can we move all this quarantine error handling into the `quarantine` function in pbench-base.sh, and then have that error handling exit the process entirely when errors are encountered?